src/components/Media/SFW.tsx

```diff
@@ -0,0 +1,56 @@
+import { Box, BoxProps } from '@mantine/core';
+import { useSession } from 'next-auth/react';
+import React from 'react';
+import { MediaCount } from './MediaCount';
+import { SfwCtx, MediaTypes } from './sfwContext';
+import { useSfwStore } from './sfwStore';
+import { MediaNsfwToggle } from './MediaNsfwToggle';
+import { SfwContent } from './sfwContent';
+import { SfwToggle } from './SfwToggle';
+import { SfwPlaceholder } from './SfwPlaceholder';
+
+type SFWProps = {
+  nsfw?: boolean;
+  type: MediaTypes;
+  id: number;
+  children?:
+    | React.ReactNode
+    | (({ nsfw, showNsfw }: { nsfw: boolean; showNsfw: boolean }) => React.ReactNode);
+};
+
+export function SFW({
+  nsfw = false,
+  type,
+  id,
+  children,
+  sx = {},
+  ...props
+}: SFWProps & Omit<BoxProps, 'children'>) {
+  const { data: session } = useSession();
```

Comment:
I created a custom hook to get the user directly from the session since we don't use the session for anything else than getting the user. This is just nitpicking so we don't have to do `session?.user?.username`, we can just do `user?.username` 😛 
```suggestion
  const user = useCurrentUser();
```